pkg/controller/master-controller-manager/master-constraint-controller/controller.go

```diff
@@ -0,0 +1,232 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package masterconstraintsyncercontroller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+
+	kubermaticapiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	controllerutil "k8c.io/kubermatic/v2/pkg/controller/util"
+	"k8c.io/kubermatic/v2/pkg/controller/util/predicate"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	kuberneteshelper "k8c.io/kubermatic/v2/pkg/kubernetes"
+	"k8c.io/kubermatic/v2/pkg/provider"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	// This controller syncs the kubermatic constraint on the master cluster to the seed clusters.
+	controllerName = "master_constraint_syncer_controller"
+)
+
+type reconciler struct {
+	log              *zap.SugaredLogger
+	recorder         record.EventRecorder
+	masterClient     ctrlruntimeclient.Client
+	namespace        string
+	seedClientGetter provider.SeedClientGetter
+}
+
+func Add(ctx context.Context,
+	mgr manager.Manager,
+	log *zap.SugaredLogger,
+	namespace string,
+	seedKubeconfigGetter provider.SeedKubeconfigGetter) error {
+
+	r := &reconciler{
+		log:              log.Named(controllerName),
+		masterClient:     mgr.GetClient(),
+		namespace:        namespace,
+		seedClientGetter: provider.SeedClientGetterFactory(seedKubeconfigGetter),
+		recorder:         mgr.GetEventRecorderFor(controllerName),
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{
+		Reconciler: r,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %v", err)
+	}
+
+	// Watch for changes to Constraints
+	if err = c.Watch(
+		&source.Kind{Type: &kubermaticv1.Constraint{}},
+		&handler.EnqueueRequestForObject{},
+		predicate.ByNamespace(namespace)); err != nil {
+		return fmt.Errorf("failed to create watch for the Constraints %v", err)
+	}
+
+	if err := c.Watch(
+		&source.Kind{Type: &kubermaticv1.Seed{}},
+		enqueueAllConstraint(r.masterClient, r.log),
+		predicate.ByNamespace(namespace),
+	); err != nil {
+		return fmt.Errorf("failed to create seed watcher: %v", err)
+	}
+
+	return nil
+}
+
+// Reconcile reconciles the kubermatic constraint on the master cluster to all seed clusters
```

Comment:
```suggestion
// Reconcile reconciles the kubermatic gatekeeper constraints in the master cluster and syncs them to all seeds
```

Reply:
I would argue to keep this as just "kubermatic constraints" (our CRD) as in all docs I use this prefix to distinguish them from the "gatkeeper constraints"